File: BugTracker.API/Models/Ticket.cs
using System;
using System.Collections.Generic;

namespace BugTracker.API.Models
{
    public class Ticket
    {
        public int Id { get; set; } 
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime Created { get; set; }
        public DateTime Updated { get; set; }
        public DateTime SubmissionDate { get; set; }
        public bool isArchived { get; set; }
        public bool isDeveloperPassed { get; set; }
        public bool isManagerPassed { get; set; }

        public Project project { get; set; }
        
        #nullable enable
        public User? User { get; set; }
        public ICollection<Comment>? Comments { get; set; }
        public Category? Category { get; set; }
        public Priority? Priority { get; set; }
        public Status? Status { get; set; }
        
    }
}

File: BugTracker-SPA/src/app/_models/Ticket.ts
import { Project } from './Project';
import { User } from './User';
import { Category } from './Category';
import { Priority } from './Priority';
import { Status } from './Status';

export interface Ticket {
    id: number;
    title: string;
    description: string;
    created: Date;
    updated: Date;
    submissionDate?: Date;
    isArchived: boolean;
    isDeveloperPassed: boolean;
    isManagerPassed: boolean;

    projectName: string;
    project: Project;
    projectId: number;

    userName: string;
    user?: User;
    userId: number;

    comments: Comment[];
    category?: Category;
    priority?: Priority;
    status?: Status;


}
